whylog/config/rule.py

```diff
@@ -32,6 +33,29 @@ def get_causes_parsers(self):
     def get_effect_name(self):
         return self._effect.name
 
+    def _gather_causes_frequency_information(self):
+        causes_with_frequency_info = [(self._causes[0].name, 0)]
+        for cause in self._causes:
```

Comment:
I'm afraid that with `Counter` I'll lose order. In my functions for input causes list:
`['a', 'a', 'b', 'c', 'c', 'c']` we got: `[('a', 2), ('b', 1), ('c', 3)]`. `Counter` does almost the same, but it returns a dictionary. I want to keep order, so thats why I keep results in list of tuples.

In other words: information keeped by this functions result is: at the begining we have 2 parsers called 'a', then one parser 'b', and finally at the end we have 3 parsers 'c'. It's helpful in `_clues_combinations`, where thanks to that information I produce combinations without repetitions.

I agree that this solution based on some additional assumptions and I want to discuss it. If causes list would look like: `['a', 'b', a']` function would return: `[('a', 1), ('b', 1), ('a', 1)]`. Function `_clues_combinations` for this data returns the same results as the older version of this function, so it produces unwanted repetitions again - because line that matched to first 'a' also matched to parser represented by second 'a' (it is the same parser, and clues list for its is the same). So the only assumption I need is that such situations will not occur.

I should'nt shuffle parsers in causes, because constraints based on these parsers order. So I think that we can shuffle this once while rule is creating.